File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()
        ans = nums[k-1] - nums[0]
        i = 0
        for j in range(k-1, len(nums)):
            ans = min(ans, nums[j] - nums[i])
            i += 1
        return ans

File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.c


int minimumDifference(int* nums, int numsSize, int k) {
    int temp;
    for (int i = 0 ; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            if (nums[i] > nums[j]) {
                temp = nums[j];
                nums[j] = nums[i];
                nums[i] = temp;     
            }
        }
    }
    int ans = nums[k-1] - nums[0];
    int i = 0;
    for (int j = k -1; j < numsSize; j++) {
        ans = ans > nums[j] - nums[i] ? nums[j] - nums[i] : ans;
        i++;
        //printf("%d %d\n", i, j);
    }
    return ans;
}